src/dune_rules/dune_file.ml

```diff
@@ -944,10 +944,88 @@ module Plugin = struct
 end
 
 module Install_conf = struct
+  module File_entry = struct
+    type t =
+      | Binding of File_binding.Unexpanded.t
+      | Include of
+          { context : Univ_map.t
+          ; path : String_with_vars.t
+          }
+
+    let decode =
+      let open Dune_lang.Decoder in
+      let decode_binding =
+        let+ binding = File_binding.Unexpanded.decode in
+        Binding binding
+      in
+      let decode_include =
+        sum
+          [ ( "include"
+            , let+ () = Syntax.since Stanza.syntax (3, 5)
+              and+ context = get_all
+              and+ path = String_with_vars.decode in
+              Include { context; path } )
+          ]
+      in
+      decode_binding <|> decode_include
```

Comment:
I think that it works, but will give a wrong hint in the case where `lang dune` < 3.5 and the `(include)` form is found (you can add a test to show that). I believe that the solution to theat is to parse using `<|>` without any `since` in the `decode_include` branch, but to call `since` depending on the shape of the result. This will correctly point that `include` is available since 3.5.

Reply:
(see the diff in `file_binding.ml` for an example of this pattern)

Reply:
There's already a test for this in test/blackbox-tests/test-cases/install-include/install-include.t:
```
  File "dune", line 5, characters 14-32:
  5 |  (files a.txt (include foo.sexp))
                    ^^^^^^^^^^^^^^^^^^
  Error: 'include' is only available since version 3.5 of the dune language.
  Please update your dune-project file to have (lang dune 3.5).
```
Does that look like the right error message?